qualcommax: ipq807x: add support for Linksys HomeWRK

testuser7 opened the pull request:
Hardware specification:
========
SoC: Qualcomm IPQ8174
Flash: 1GB (Micron MT29F8G08ABBCAH4 or AMD/Spansion S34MS08G2)
RAM: 2GB (2x Kingston B5116ECMDXGJD or ESMT M15T2G16128A DDR3L)
Ethernet: 4x 10/100/1000Mbps (Qualcomm QCA8075)
WiFi1: 5GHz ax 2x2 (Qualcomm QCN5054 + Skyworks SKY85755-11) - channels 36-64 (low band)
WiFi2: 2.4GHz ax 2x2 (Qualcomm QCN5024 + Skyworks SKY85340-11)
WiFi3: 5GHz ax 4x4 (Qualcomm QCN5054 + Skyworks SKY85755-11) - channels 100-177 (high band)
LED: 1x RGB status (NXP PCA9633)
USB: 1x USB 3.0
Button: WPS, Reset

Flash instructions:
========
1. Installation from initramfs image using USB drive:
    Put the initramfs image on the USB drive:
    - dd bs=1M if=openwrt-qualcommax-ipq807x-linksys_homewrk-initramfs-uImage.itb of=/dev/sdX

    Connect serial console (JST PH 6-pin connector):
    - 1 - GND, 2 - TX, 4 - RX (115200, 8N1, 3.3V)

    Stop u-boot and run:
    - usb start && usbboot 44000000 0 && bootm 44000000

    Write firmware to the flash from initramfs:
    - sysupgrade -n openwrt-qualcommax-ipq807x-linksys_homewrk-squashfs-sysupgrade.bin

    or
    - ubiformat /dev/mtd22 -y -f openwrt-qualcommax-ipq807x-linksys_homewrk-squashfs-factory.ubi

3. USB recovery:
    Put the initramfs image on the USB:
    - dd bs=1M if=openwrt-qualcommax-ipq807x-linksys_homewrk-initramfs-uImage.itb of=/dev/sdX

    Set u-boot env:
    - fw_setenv bootusb 'usb start && usbboot 44000000 0 && bootm 44000000'
    - fw_setenv bootcmd 'run bootusb; bootipq'

testuser7 committed:
qualcommax: ipq807x: add support for Linksys HomeWRK

Hardware specification:
========
SoC: Qualcomm IPQ8174
Flash: 1GB (Micron MT29F8G08ABBCAH4 or AMD/Spansion S34MS08G2)
RAM: 2GB (2x Kingston B5116ECMDXGJD or ESMT M15T2G16128A DDR3L)
Ethernet: 4x 10/100/1000Mbps (Qualcomm QCA8075)
WiFi1: 5GHz ax 2x2 (Qualcomm QCN5054 + Skyworks SKY85755-11) - channels 36-64 (low band)
WiFi2: 2.4GHz ax 2x2 (Qualcomm QCN5024 + Skyworks SKY85340-11)
WiFi3: 5GHz ax 4x4 (Qualcomm QCN5054 + Skyworks SKY85755-11) - channels 100-177 (high band)
LED: 1x RGB status (NXP PCA9633)
USB: 1x USB 3.0
Button: WPS, Reset

Flash instructions:
========
1. Installation from initramfs image using USB drive:
Put the initramfs image on the USB drive:
- dd bs=1M if=openwrt-qualcommax-ipq807x-linksys_homewrk-initramfs-uImage.itb of=/dev/sdX
Connect serial console (JST PH 6-pin connector):
- 1 - GND, 2 - TX, 4 - RX (115200, 8N1, 3.3V)
Stop u-boot and run:
- usb start && usbboot 44000000 0 && bootm 44000000
Write firmware to the flash from initramfs:
- sysupgrade -n openwrt-qualcommax-ipq807x-linksys_homewrk-squashfs-sysupgrade.bin
or
- ubiformat /dev/mtd22 -y -f openwrt-qualcommax-ipq807x-linksys_homewrk-squashfs-factory.ubi

2. USB recovery:
Put the initramfs image on the USB:
- dd bs=1M if=openwrt-qualcommax-ipq807x-linksys_homewrk-initramfs-uImage.itb of=/dev/sdX
Set u-boot env:
- fw_setenv bootusb 'usb start && usbboot 44000000 0 && bootm 44000000'
- fw_setenv bootcmd 'run bootusb; bootipq'

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17463
Signed-off-by: Robert Marko <robimarko@gmail.com>

testuser7 commented:
Is there any difference between installing a ubi factory image via `mtd` and `ubiformat`?
- `mtd -e rootfs -n write openwrt-qualcommax-ipq807x-linksys_homewrk-squashfs-factory.ubi rootfs`
- `ubiformat /dev/mtd22 -y -f openwrt-qualcommax-ipq807x-linksys_homewrk-squashfs-factory.ubi`

robimarko commented:
If its UBI image you should use ubiformat as it will scan for bad blocks first as well as reset the counter to 0

robimarko commented on target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8174-homewrk.dts:
```diff
@@ -0,0 +1,68 @@
+	};
+
+	chosen {
+        bootargs-append = " root=/dev/ubiblock0_1";
```
Broken indent

testuser7 committed:
uboot-envtools: ipq807x: remove number of blocks and code simplification

Envtools can automatically detect the number of blocks.

Signed-off-by: Paweł Owoc <frut3k7@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/17463
Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented on package/boot/uboot-envtools/files/qualcommax_ipq807x:
```diff
@@ -29,6 +29,11 @@ edimax,cax1800)
+linksys,homewrk)
+	idx="$(find_mtd_index 0:appsblenv)"
+	[ -n "$idx" ] && \
+		ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x40000" "0x40000" "1"
```
Yeah, and its working on the Dynalink after removing `/etc/config/ubootenv` so it gets regenerated.

testuser7 replied:
Done.

robimarko commented:
Can you just bump ipq-wifi since I merged the BDF and I think this is ready to go

testuser7 commented:
> Can you just bump ipq-wifi since I merged the BDF and I think this is ready to go

Done.

robimarko commented:
Thanks! Rebased on top of main and merged!
